### PR TITLE
chore: bump k8s support to 1.29 for v0.34.x

### PR DESCRIPTION
**Pull request by jmdeal**
Fixes #N/A 

**Description**
Bumps the supported kubernetes version for v0.34.x to 1.29. Additionally, updates the docgen to include all supported versions (1.23 - 1.29). 
**How was this change tested?**
N/A

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jmdeal** [comment]
Blocked until the following PR is merged into upstream: https://github.com/kubernetes-sigs/karpenter/pull/569

File: hack/docgen.sh
Comment (jonathan-innis) [inline hack/docgen.sh:6]: musing: What does this number mean again? Would be nice to use CLI args so it's a little easier to reason about

File: hack/docs/compatibility-karpenter.yaml
Comment (jonathan-innis) [inline hack/docs/compatibility-karpenter.yaml:45]: Do we need to add this version or does this get auto-added as part of the release process?

**Review by jonathan-innis — APPROVED** [review]
Couple questions, optional though

File: hack/docgen.sh
Comment (jmdeal) [reply hack/docgen.sh:6]: The number of supported k8s versions. This makes sure we still include 1.23 rather than starting at 1.24.

File: hack/docs/compatibility-karpenter.yaml
Comment (jmdeal) [reply hack/docs/compatibility-karpenter.yaml:45]: You're right, I'll limit this to just the CI changes and not the actual updates.

File: hack/docgen.sh
Comment (engedaam) [reply hack/docgen.sh:6]: Can at least make the value a variable to give use some context?

Comment (jmdeal) [reply hack/docgen.sh:6]: Sure, that's easy enough. Probably should update all of the hack programs to use named flags at some point. I'm imagining just `--file` and `--version-count` here.

**Review by engedaam — DISMISSED** [review]
LGTM 🚀

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Comment by dntosas** [comment]
is this worth a quick release? wdyt?

**Pull request merged**
